xgcm/test/test_metrics.py

```diff
@@ -138,6 +136,105 @@ def test_get_metric(axes, data_var, drop_vars, metric_expected_list, expected_er
         assert metric.equals(expected)
 
 
+@pytest.mark.parametrize(
+    "set_metrics,metric_axes,ds_array,ds_expected,set_metric_vars,expected_metric",
```

Comment:
This is very impressive use of `parametrize` but it's a hard to follow because your parameters and the actual test code are far away from each other. 

```
if expected_metric == "cond1":
```
I think we should break this test up in to 4 tests where each test tests a single condition. There's not that much code shared between the 4 actual tests so there's not much gained by sticking it in one function.